### PR TITLE
Initial setup for url mappings feature

### PR DESCRIPTION
**Pull request by monu-k2io**
New application url mappings POJO
New agent APIs for url mappings
latest gradle file

### CODE REVIEW

File: newrelic-security-agent/src/main/java/com/newrelic/agent/security/instrumentator/utils/AgentUtils.java [hunk 1]
```diff
@@ -644,4 +646,10 @@ private void applyNRPolicyOverride() {
         this.setPolicyOverridden(override);
     }
 
+
+    public static void sendApplicationURLMappings() {
+        ApplicationURLMappings applicationURLMappings = new ApplicationURLMappings(URLMappingsHelper.getApplicationURLMappings());
+        logger.logInit(LogLevel.INFO, String.format("Collected application url mappings %s", applicationURLMappings.toString()), Agent.class.getName());
```
Comment (lovesh-ap) [inline newrelic-security-agent/src/main/java/com/newrelic/agent/security/instrumentator/utils/AgentUtils.java:652]: .toString() is not required.

File: settings.gradle [hunk 2]
```diff
@@ -134,4 +137,14 @@ include 'instrumentation:jdbc-inet-merlia'
...
+include 'instrumentation:jersey'
+include 'instrumentation:resteasy-2.2'
+include 'instrumentation:resteasy-3'
+include 'instrumentation:resteasy-4'
+include 'instrumentation:apache-wicket-6.4'
+include 'instrumentation:apache-wicket-7.0'
+include 'instrumentation:apache-wicket-8.0'
+include 'instrumentation:grails-1.3'
+include 'instrumentation:grails-2.0'
+include 'instrumentation:grails-3.0'
```
Comment (lovesh-ap) [inline settings.gradle:150]: The above mention modules are missing from code base in this PR, hence the build will fail. Please remove them.

**Review by lovesh-ap — APPROVED** [review]

File: newrelic-security-agent/src/main/java/com/newrelic/agent/security/instrumentator/utils/AgentUtils.java (hunk 1 above)
Comment (IshikaDawda) [reply newrelic-security-agent/src/main/java/com/newrelic/agent/security/instrumentator/utils/AgentUtils.java]: Removed this in commit [710fb2b](https://github.com/newrelic/csec-java-agent/pull/51/commits/710fb2bc4bcb8d35cf9db31503e88d3aa3ce1761)

File: newrelic-security-api/src/main/java/com/newrelic/api/agent/security/instrumentation/helpers/URLMappingsHelper.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import java.util.concurrent.ConcurrentHashMap;
+
+public class URLMappingsHelper {
+    private static Set<ApplicationURLMapping> mappings = ConcurrentHashMap.newKeySet();
+
+    public static Set<ApplicationURLMapping> getApplicationURLMappings() {
+        return mappings;
+    }
+
+    public static void addApplicationURLMapping(ApplicationURLMapping obj) {
```
[the remark this answers is not in the document]
Comment (IshikaDawda) [reply newrelic-security-api/src/main/java/com/newrelic/api/agent/security/instrumentation/helpers/URLMappingsHelper.java]: Updated variable name in commit [9857ffa](https://github.com/newrelic/csec-java-agent/pull/51/commits/9857ffafd07c9ddf29c145b1b2213b6869e77c8e)

File: settings.gradle (hunk 2 above)
Comment (IshikaDawda) [reply settings.gradle]: Removed unnecessary module names in commit [0add19a](https://github.com/newrelic/csec-java-agent/pull/51/commits/0add19a2c8698f3de7b6613647780482163d7e5b)

**Pull request merged**